Geometry/HGCalCommonData/interface/HGCalCellUV.h

```diff
@@ -0,0 +1,50 @@
+#ifndef Geometry_HGCalCommonData_HGCalCellUV_h
+#define Geometry_HGCalCommonData_HGCalCellUV_h
+
+#include <cmath>
+#include <cstdint>
+#include <iterator>
+#include <map>
+#include "Geometry/HGCalCommonData/interface/HGCalCell.h"
+
+class HGCalCellUV {
+public:
+  HGCalCellUV(double waferSize, double separation, int32_t nFine, int32_t nCoarse);
+
+  std::pair<int32_t, int32_t> cellUVFromXY1(
+      double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug);
+
+  std::pair<int32_t, int32_t> cellUVFromXY2(
+      double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug);
+
+  std::pair<int32_t, int32_t> cellUVFromXY3(
+      double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug);
+
+  std::pair<int32_t, int32_t> cellUVFromXY4(
+      double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug);
+
+private:
+  std::pair<int32_t, int32_t> cellUVFromXY4(double xloc,
+                                            double yloc,
+                                            int ncell,
+                                            double cellX,
+                                            double cellY,
+                                            double cellXTotal,
+                                            double cellYTotal,
+                                            std::map<std::pair<int, int>, std::pair<double, double> >& cellPos,
+                                            bool extend,
+                                            bool debug);
+
+  const double sqrt3_ = std::sqrt(3.0);
+  const double sqrt3By2_ = (0.5 * sqrt3_);
+  const double sin60_ = sqrt3By2_;
+  const double cos60_ = 0.5;
```

Comment:
Another simplification could be to use everywhere in the code `sin60_` instead of `sqrt3By2_`, since this is everywhere what it is ment to (as well as using `cos60_` instead of 0.5 when it has the same meaning):
```suggestion
  const double sin60_ = 0.5 * sqrt3_;
  const double cos60_ = 0.5;
```
But let it for a possible further code cleaning